torch/distributed/fsdp/_exec_order_utils.py

```diff
@@ -135,15 +133,15 @@ def get_handles_to_forward_prefetch(
         if current_index is None:
             return None
         target_index = current_index + 1
-        target_handles_keys: List[_HandlesKey] = []
+        target_handles_key: List[FlatParamHandle] = None
         for _ in range(self._forward_prefetch_limit):
             if target_index >= len(self.handles_pre_forward_order):
                 break
-            target_handles_keys.append(self.handles_pre_forward_order[target_index])
+            target_handles_key = self.handles_pre_forward_order[target_index]
             target_index += 1
-        return target_handles_keys
+        return target_handles_key
 
-    def record_post_forward(self, handles: List[FlatParamHandle]) -> None:
+    def record_post_forward(self, handle: FlatParamHandle) -> None:
```

Comment:
```suggestion
    def record_post_forward(self, handle: Optional[FlatParamHandle]) -> None:
```
since we may actually pass `state._handle` directly, which may be `None`